janusgraph-berkeleyje/src/main/java/org/janusgraph/diskstorage/berkeleyje/BerkeleyJETx.java

```diff
@@ -50,8 +53,28 @@ public Transaction getTransaction() {
         return tx;
     }
 
-    private void closeOpenIterators() {
-        openCursors.forEach(Cursor::close);
+    Cursor openCursor(Database db) throws BackendException {
+        synchronized (openCursors) {
+            if (!isOpen) {
+                throw new PermanentBackendException("Transaction is not valid");
```

Comment:
(nitpick) I think it would be helpful for users if the message would include the reason why the transaction is not valid which is that it's already closed in this case.

Reply:
Done